Essentials/src/com/earth2me/essentials/ReserveEssentials.java

```diff
@@ -0,0 +1,1830 @@
+package com.earth2me.essentials;
+
+import com.earth2me.essentials.api.NoLoanPermittedException;
+import com.earth2me.essentials.api.UserDoesNotExistException;
+import net.ess3.api.Economy;
+import net.tnemc.core.Reserve;
+import net.tnemc.core.economy.EconomyAPI;
+import net.tnemc.core.economy.currency.Currency;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+
+import java.math.BigDecimal;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static com.earth2me.essentials.I18n.tl;
+
+/**
+ * @author creatorfromhell <daniel.viddy@gmail.com>
+ */
+public class ReserveEssentials implements EconomyAPI {
+
+    private Essentials plugin;
+
+    public ReserveEssentials(Essentials plugin) {
+        this.plugin = plugin;
+    }
+
+    public static void register(Essentials plugin) {
+        //Check to see if there is an economy provider already registered, if so Essentials will take the back seat.
+        if(!((Reserve)Bukkit.getServer().getPluginManager().getPlugin("Reserve")).economyProvided()) {
+            Reserve.instance().registerProvider(new ReserveEssentials(plugin));
+        }
+    }
+
+    /**
+     * @return The name of the Economy implementation.
+     */
+    @Override
+    public String name() {
+        return "Essentials Economy";
```

Comment:
```suggestion
        return "EssentialsX";
```

"Economy" seems redundant, and this is specifically EssentialsX.